bitcoin/src/taproot.rs

```diff
@@ -603,17 +924,22 @@ impl ScriptLeaf {
         u8::try_from(self.merkle_branch.0.len()).expect("depth is guaranteed to fit in a u8")
     }
 
-    /// Computes a leaf hash for this [`ScriptLeaf`].
+    /// Computes a leaf hash for this [`ScriptLeaf`] if the leaf is known.
     #[inline]
-    pub fn leaf_hash(&self) -> TapLeafHash { TapLeafHash::from_script(&self.script, self.ver) }
+    pub fn leaf_hash(&self) -> Option<TapLeafHash> {
```

Comment:
If the leaf is not known we already know the hash and should return it?

Reply:
I thought it might be confusing because the hidden node might not necessarily be a leaf. It could be a hidden tree.

Reply:
I had a note for adding `node_hash()` method that returns the `TapNodeHash`

Reply:
Ah, that makes sense. Agree with adding `node_hash`.